src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.DefaultItems.props

```diff
@@ -37,4 +37,8 @@ Copyright (c) .NET Foundation. All rights reserved.
     <PackageReference Include="Microsoft.NETCore.App" Version="$(RuntimeFrameworkVersion)" IsImplicitlyDefined="true" />
   </ItemGroup>
 
+  <ItemGroup Condition=" '$(DisableImplicitFrameworkReferences)' != 'true' and '$(TargetFrameworkIdentifier)' == '.NETCoreApp' and !$(RuntimeFrameworkVersion.StartsWith('1.'))">
+    <PackageReference Include="Microsoft.NETCore.DotNetAppHost" Version="$(RuntimeFrameworkVersion)" IsImplicitlyDefined="true" />
```

Comment:
Status quo for package referencing is as follows:

M.N.App -> HostPolicy -> HostResolver -> Dotnet

In the above, the latter three may not ever change once released. Thus, when Core-Setup build infrastructure publishes sharedFX, it has the need to restore the versions of the latter three packages. This is done via a special mechanism where this version information is maintained and this allows dotnet.exe released with 1.0.X (for example) to be published as part of M.N.App 1.0.5 and also being included in the respective platform installer.

Simply removing DotnetHost package or changing its contents will break this approach.

Moving forward to the AppHost work, @schellap  added AppHost package dependency to HostFXR package (see https://github.com/dotnet/core-setup/commit/ea86ceed56522a0701dcd6e8966bc3407baab4ed#diff-4482ab3edd31393e59c026dd6f9d60f6R19) and updated the SharedFX publisher task to delete AppHost package (https://github.com/dotnet/core-setup/blob/ea86ceed56522a0701dcd6e8966bc3407baab4ed/build_projects/shared-build-targets-utils/Utils/SharedFrameworkPublisher.cs#L130).

Hope this explains.